pkg/controller/clusterpool/clusterpool_controller_test.go

```diff
@@ -944,26 +960,25 @@ func TestReconcileClusterPool(t *testing.T) {
 				assert.Equal(t, test.expectedObservedReady, pool.Status.Ready, "unexpected observed ready count")
 			}
 
-			missingDependentsCondition := controllerutils.FindClusterPoolCondition(pool.Status.Conditions, hivev1.ClusterPoolMissingDependenciesCondition)
-			if test.expectedMissingDependenciesStatus != nil {
-				require.NotNil(t, missingDependentsCondition)
-				expectedStatus := corev1.ConditionFalse
-				if *test.expectedMissingDependenciesStatus {
-					expectedStatus = corev1.ConditionTrue
+			if test.expectedMissingDependenciesStatus != "" {
+				missingDependenciesCondition := controllerutils.FindClusterPoolCondition(pool.Status.Conditions, hivev1.ClusterPoolMissingDependenciesCondition)
+				if assert.NotNil(t, missingDependenciesCondition, "did not find MissingDependencies condition") {
```

Comment:
Since we always expect missingDependenciesCondition in any case, shouldn't this be `require.NotNil` ? I believe require is a precondition here

Reply:
assert returns a boolean - so it would compare the status only if it missingDependencies is present. I prefer the assert because it allows to proceed even if the assert here returns false

Reply:
If everything below depended on the missingDependenciesCondition I'd say require, but it looks like we transition to other checks so assert is probably ok here and lets us get more info out of the test failure.